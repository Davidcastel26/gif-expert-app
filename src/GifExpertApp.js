import React from 'react'
import PropTypes from 'prop-types'

const GifExpertApp = (props) => {

    const pikey = 'AYhFpbImnSFFHIwxeoq82SNAeF5iDQXX'

    return (
        <>
            <h2>Gif Expert App</h2>
            <hr />
        </>
    )
}

GifExpertApp.propTypes = {

}

export default GifExpertApp

