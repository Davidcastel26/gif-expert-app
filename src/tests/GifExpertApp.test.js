//react imports
import React from 'react';
import GifExpertApp from '../GifExpertApp';

// jest imports 
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('testing the primary react component', () => {

    const wrapper = shallow( <GifExpertApp /> )

    test('should show the right path', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should show a list of categories', () => {

        const categoriesH = ['Arcane'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categoriesH}/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categoriesH.length )

    })

})
