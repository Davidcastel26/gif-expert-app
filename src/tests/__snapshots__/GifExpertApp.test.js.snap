// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the primary react component should show a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Arcane"
      key="Arcane"
    />
  </ol>
</Fragment>
`;

exports[`testing the primary react component should show the right path 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
